src/org/labkey/test/components/ui/grids/DetailTableEdit.java

```diff
@@ -351,6 +355,64 @@ public DetailTableEdit clearSelectValue(String fieldCaption, boolean waitForSele
         return this;
     }
 
+    /**
+     * Set a DateTime, Date or Time field.
+     * @param fieldCaption The caption of the field to set.
+     * @param dateTime Will be used to determine what kind of field is being set and how to set it. If the parameter
+     *                 is a LocalDateTime object then it is assumed that field is a DateTime field. If the parameter is
+     *                 a LocalDate object then it is assumed to be a date-only field. And I think you can guess what
+     *                 happens with a LocalTime object type. If the type is a string it is used as a literal value that
+     *                 is typed into the field (no picker is used).
+     * @return A reference to this DetailTableEdit object.
+     */
+    public DetailTableEdit setDateTimeField(String fieldCaption, Object dateTime)
+    {
+        ReactDateTimePicker dateTimePicker = getDateTimePicker(fieldCaption);
+        if(dateTime instanceof LocalDateTime localDateTime)
+        {
+            dateTimePicker.select(localDateTime);
+        }
+        else if(dateTime instanceof LocalDate localDate)
+        {
+            dateTimePicker.selectDate(localDate);
+        }
+        else if(dateTime instanceof LocalTime localTime)
+        {
+            dateTimePicker.selectTime(localTime);
+        }
+        else if(dateTime instanceof String setValue)
+        {
+            dateTimePicker.set(setValue, true);
+            // When setting DateTime, Date-only or Time-only fields by text there is a slight delay as any formatting is applied.
+            getWrapper().sleep(500);
```

Comment:
Should this sleep (or some sort of wait) be baked in to `ReactDateTimePicker.set`?

Reply:
I'll take a look. I can see where it might be easier to check for the update in that component than here.